extensions/s3/src/main/java/io/deephaven/extensions/s3/S3CompletableOutputStream.java

```diff
@@ -309,42 +314,37 @@ private void sendPartRequest(final S3WriteRequest request) throws IOException {
                 .build();
         request.buffer.flip();
         request.partNumber = nextPartNumber;
-        request.sdkUploadFuture =
-                s3AsyncClient.uploadPart(uploadPartRequest, AsyncRequestBody.fromByteBufferUnsafe(request.buffer));
-        request.completionFuture = request.sdkUploadFuture
+        request.sdkUploadFuture = s3AsyncClient.uploadPart(uploadPartRequest,
+                AsyncRequestBody.fromByteBufferUnsafe(request.buffer))
```

Comment:
You are chaining a new future, we need to capture the original from the library call.